Preact: Optimize chat room rendering performance

DieterReinert opened the pull request:
Optimize chat room opening and rendering performance with requestAnimationFrame batching and component structure improvements.

Measured with Chrome DevTools performance traces on lobby (465+ users):

| Metric | Before | After | Improvement |
|--------|--------|-------|-------------|
| **Trace Events** | 74,458 | 22,303 | **70% reduction** |
| **Longest Task** | 297ms | 185ms | **38% faster** |

DieterReinert committed:
Preact: Optimize chat room rendering performance

Zarel commented:
I like the `requestAnimationFrame` changes, but I don't think `UserItem` is going to improve anything (readability or perf). As for `VirtualUserList`, I can imagine `minimized` improving things, but I don't think making it a component improves anything else.

DieterReinert committed:
Simplify chat user list rendering

- inline user button rendering in ChatUserList
- skip mapping when minimized to avoid needless work
